sql/inspect: clarify error message for internal failures

spilchen opened the pull request:
Previously, when `INSPECT` encountered internal errors, it always reported "INSPECT found inconsistencies," even if the issue was caused by something other than actual data corruption. This made it hard to tell whether the problem was in user data or within INSPECT itself (for example, a bad query generated internally).

This commit refines that behavior. If all observed errors are internal, `INSPECT` now reports: "INSPECT encountered internal errors." This makes it clear that the problem might stem from an internal failure in INSPECT or from data corruption detected during internal queries, rather than always implying user data inconsistencies.

Additionally, a hint is now included in the error message, regardless of error type, guiding users to run SHOW INSPECT ERRORS to retrieve more information.

Informs: #155676
Epic: CRDB-55075

Release note: none

rafiss commented on pkg/sql/inspect/inspect_job.go:
```diff
@@ -105,8 +105,9 @@ func (c *inspectResumer) OnFailOrCancel(
-	if jobErr != nil && errors.Is(jobErr, errInspectFoundInconsistencies) {
+	// Record RunsWithIssues metric if the job failed due to finding issues (including internal errors).
+	if jobErr != nil &&
+		(errors.Is(jobErr, errInspectFoundInconsistencies) || errors.Is(jobErr, errInspectInternalErrors)) {
```
super nit: `errors.Is` already has a nil check, so we can skip that explicit check here.

spilchen replied:
Done.
<!-- Reviewable comment -Oc6fGzfBDtLhdIunYVn:b-896fix -->
<!-- Sent from Reviewable.io -->

spilchen committed:
sql/inspect: clarify error message for internal failures

Previously, when INSPECT encountered internal errors, it always reported
"INSPECT found inconsistencies," even if the issue was caused by
something other than actual data corruption. This made it hard to tell
whether the problem was in user data or within INSPECT itself (for
example, a bad query generated internally).

This commit refines that behavior. If all observed errors are internal,
INSPECT now reports: "INSPECT encountered internal errors." This makes
it clear that the problem might stem from an internal failure in INSPECT
or from data corruption detected during internal queries, rather than
always implying user data inconsistencies.

Additionally, a hint is now included in the error message, regardless
of error type, guiding users to run SHOW INSPECT ERRORS to retrieve
more information.

Informs: #155676
Epic: CRDB-55075

Release note: none

spilchen commented on pkg/sql/inspect/inspect_logger.go:
```diff
@@ -23,29 +23,56 @@ type inspectLogger interface {
+// tracking aggregate state about the issues encountered.
+type inspectLoggerBundle struct {
+	loggers             []inspectLogger
+	foundIssue          atomic.Bool
```
Good idea.
<!-- Reviewable comment -Oc6frf53N5X-drC5q05:b-ovwaw7 -->
<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/155766)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @spilchen)



<!-- Sent from Reviewable.io -->

rafiss reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/155766#-:-OcCPelk1Mqkw3iEd71u:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->
